tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_volume_scenarios.py

```diff
@@ -0,0 +1,199 @@
+        # Get the client nodes
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        # Authenticate the clients
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        # Prepare the clients
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        ran_string = "".join(random.choices("abcdefghijklmnopqrstuvwxyz", k=5))
+        start_volume = f"cephfs1_{ran_string}"
+        client1.exec_command(sudo=True, cmd=f"ceph fs volume create {start_volume}")
+        log.info(client1.exec_command(sudo=True, cmd="ceph fs ls"))
+        volume_name_list = []
+        # Try to delete the volume with mon_allow_pool_delete to false
+        client1.exec_command(
+            sudo=True, cmd="ceph config set mon mon_allow_pool_delete false"
+        )
+        delete_result, delete_ec = client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs volume rm {start_volume} --yes-i-really-mean-it",
+            check_ec=False,
+        )
+        if delete_result == 0:
+            log.error(
+                "Volume deletetion should not succeed when mon_allow_pool_delete is false"
+            )
+            return 1
+        log.info("Volume deletion failed as expected")
+        # Set mon_allow_pool_delete to true
+        client1.exec_command(
+            sudo=True, cmd="ceph config set mon mon_allow_pool_delete true"
+        )
+        # Delete the volume and check if the volume is deleted
+        delete_result2, delete_ec2 = client1.exec_command(
+            sudo=True, cmd=f"ceph fs volume rm {start_volume} --yes-i-really-mean-it"
+        )
+        if delete_ec2 == 0:
+            log.error("Volume deletion failed")
+            return 1
+        else:
+            log.info("Volume deletion successful")
+        # Rename the volume
+        client1.exec_command(
+            sudo=True, cmd=f"ceph fs volume create cephfs1_{ran_string}"
+        )
+        volume_name_list.append(f"cephfs1_{ran_string}")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs volume rename cephfs1_{ran_string} cephfs2_{ran_string} --yes-i-really-mean-it",
+        )
+        # check if the volume is renamed
+        rename_result, rename_ec = client1.exec_command(sudo=True, cmd="ceph fs ls")
+        if "cephfs2" not in rename_result:
+            log.error("Volume rename failed")
+            return 1
+        else:
+            log.info("Volume rename successful")
+        # Rename the volume with additional pools
+        pool_name = f"cephfs2_data_pool_{ran_string}"
+        client1.exec_command(sudo=True, cmd=f"ceph osd pool create {pool_name} 128 128")
+        client1.exec_command(
+            sudo=True, cmd=f"ceph fs add_data_pool cephfs2_{ran_string} {pool_name}"
+        )
+        volume_name_list.append(f"cephfs2_{ran_string}")
+        client1.exec_command(
```

Comment:
for this also you can add common function so that it can be used by others. If you see benifit

Reply:
do u mean for adding data_pool?